apps/api/src/app/topics/use-cases/filter-topics/filter-topics.use-case.ts

```diff
@@ -35,20 +19,40 @@ export class FilterTopicsUseCase {
       throw new BadRequestException(`Page size can not be larger then ${DEFAULT_TOPIC_LIMIT}`);
     }
 
-    const query = mapFromCommandToEntity(command);
+    const query = this.mapFromCommandToEntity(command);
 
     const totalCount = await this.topicRepository.count(query);
 
-    const data = await this.topicRepository.find(query, '', {
+    const pagination = {
       limit: pageSize,
       skip: page * pageSize,
-    });
+    };
+    const filteredTopics = await this.topicRepository.filterTopics(query, pagination);
 
     return {
       page,
       totalCount,
       pageSize,
-      data: data.map(mapFromEntityToDto),
+      data: filteredTopics.map(this.mapFromEntityToDto),
+    };
+  }
+
+  private mapFromCommandToEntity(
+    command: FilterTopicsCommand
+  ): Pick<TopicEntity, '_environmentId' | 'key' | '_organizationId'> {
+    return {
+      _environmentId: TopicRepository.convertStringToObjectId(command.environmentId),
+      _organizationId: TopicRepository.convertStringToObjectId(command.organizationId),
+      ...(command.key && { key: command.key }),
+    };
+  }
+
+  private mapFromEntityToDto(topic: TopicEntity & { subscribers: ExternalSubscriberId[] }): TopicDto {
```

Comment:
the returned type should be a partial from the `TopicDto` as we don't return subscribers

Reply:
No, the returned type is right. We are returning subscribers from:
https://github.com/novuhq/novu/blob/58bb5b321274ce7258d2fec7ba3a3203eda8d920/libs/dal/src/repositories/topic/topic.repository.ts#L54
We are returning the `Promise<TopicEntity & { subscribers: ExternalSubscriberId[] }> ` that then mapped to TopicDto that includes the `subscribers` field.
Check that we are doing in that repo method a lookup with a projection to provide the subscribers:
https://github.com/novuhq/novu/blob/58bb5b321274ce7258d2fec7ba3a3203eda8d920/libs/dal/src/repositories/topic/topic.repository.ts#L18

Reply:
ahh damn I've missed that as usual :D